monitoredresources/monitored_resources.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoredresources
+
+import (
+	"fmt"
+	"os"
+	"sync"
+)
+
+type MonitoredResInterface interface {
+	GetType() string
```

Comment:
I think these can actually be unified into one method that can also return an error, eg:
```
MonitoredResource() (type string, labels map[string]string, err error)
```

Reply:
I added MonitoredResource() (type sting, labels map[string]string) . There is no possibility of errors. Either we detect the resource or do not detect it. So error is not part of the return value.